d8/d8batch/d8batch.module

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * @file
+ * Contains d8batch.module.
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\DomCrawler\Crawler;
+use Drupal\node\Entity\Node;
+use GuzzleHttp\Client;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_help().
+ */
+function d8batch_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the d8batch module.
+    case 'help.page.d8batch':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Batch API demonstration in Drupal 8.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements callback_batch_operation().
+ *
+ * @param string $url
+ *   Feed source URL.
+ */
+function _d8batch_batch_operation($url, array $context) {
+  try {
+    $url = Url::fromUri($url);
+    if (!d8batch_check_if_feed_source_already_exists($url)) {
+      try {
+        $client = new Client();
+        $response = $client->get($url->getUri());
+        if ($response->getStatusCode() == 200) {
+          // If title can not be extracted, then use the URL as title.
+          if (!($title = _d8batch_get_title_of_feed_source($response->getBody()->getContents()))) {
+            $title = $url->getUri();
+          }
+          // Create a new node.
+          $node = Node::create([
+            'type' => 'feed_source',
+            'title' => $title,
+            'field_feed_source_url' => [
+              'uri' => $url->getUri(),
+            ],
+          ]);
+          try {
+            $node->save();
+            $context['results']['success'][] = $node;
+          }
+          catch (EntityStorageException $e) {
+            // Node can not be created.
+            watchdog_exception('d8batch', $e);
+            $context['results']['failed'][] = $url->getUri();
+          }
+        }
+        else {
+          // Wrong status code.
+          $context['results']['failed'][] = $url->getUri();
+        }
+      }
+      catch (Exception $e) {
+        // Feed source is unreachable on this url.
+        watchdog_exception('d8batch', $e);
+        $context['results']['failed'][] = $url->getUri();
+      }
+    }
+  }
+  catch (InvalidArgumentException $e) {
+    // Feed source url not valid.
+    watchdog_exception('d8batch', $e);
+    $context['results']['failed'][] = $url->getUri();
+  }
+}
+
+/**
+ * Implements callback_batch_finished().
+ */
+function _d8batch_batch_finished($success, $results, $operations) {
+  if ($success) {
+    $message = \Drupal::translation()
```

Comment:
I did not find any info about this. The official documentation uses the same solution:
https://api.drupal.org/api/drupal/core!includes!form.inc/group/batch/8